test/core/data.wast

```diff
@@ -0,0 +1,110 @@
+;; Test the data section
+
+;; Syntax
+
+(module
```

Comment:
Some of these tests seem to be duplicated in `memory.wast`, is it worth moving these tests there or removing those instead?

Reply:
Good point. I moved all data-specific tests over and added quite a few more. PTAL